tests/fs/fs.c

```diff
@@ -922,6 +922,54 @@ static void test_enotdir()
     _passed(__FUNCTION__);
 }
 
+static void test_fdatasync()
+{
+    int fd = creat("/test_fdatasync", 0666);
+    assert(fd >= 0);
+    struct stat statbuf;
+    int ret = fstat(fd, &statbuf);
+    assert(ret == 0);
+    int init_size = statbuf.st_size;
+    ret = write(fd, alpha, sizeof(alpha));
+    assert(ret == sizeof(alpha));
+    ret = fdatasync(fd);
+    assert(ret == 0);
+    close(fd);
```

Comment:
I think `close(fd)` triggers data writing back to disk. We should close the file after `fdatasync` is verified.

Reply:
Fixed.